processor/resourcedetectionprocessor/internal/aws/ec2/ec2.go

```diff
@@ -71,5 +77,63 @@ func (d *Detector) Detect(ctx context.Context) (pdata.Resource, error) {
 	attr.InsertString(conventions.AttributeHostType, meta.InstanceType)
 	attr.InsertString(conventions.AttributeHostName, hostname)
 
+	tags, err := connectAndFetchEc2Tags(meta.Region, meta.InstanceID, d.cfg)
+	if err != nil {
+		return res, fmt.Errorf("failed fetching ec2 instance tags: %w", err)
+	}
+
+	for key, val := range tags {
+		attr.InsertString(tagPrefix + key, val)
+	}
+
 	return res, nil
 }
+
+func connectAndFetchEc2Tags(region string, instanceID string, cfg Config) (map[string]string, error) {
+	if(!cfg.AddAllTags && len(cfg.TagsToAdd) == 0){
+		return nil, nil
+	}
+
+	sess, err := session.NewSession(&aws.Config{
+		Region: aws.String(region)},
+	)
+	if err != nil {
+		return nil, err
+	}
+	if _, err = sess.Config.Credentials.Get(); err != nil {
```

Comment:
Do we need this line? It will try to fetch the credentials on the first request.